pandas/core/missing.py

```diff
@@ -964,3 +974,107 @@ def _rolling_window(a: np.ndarray, window: int):
     shape = a.shape[:-1] + (a.shape[-1] - window + 1, window)
     strides = a.strides + (a.strides[-1],)
     return np.lib.stride_tricks.as_strided(a, shape=shape, strides=strides)
+
+
+def _can_hold_element(values, element: Any) -> bool:
```

Comment:
is there a version of this that makes the existing can_hold_element more accurate instead of implementing another function?

Reply:
Not fully sure what I was thinking in April / if that was already the case, but so right now the existing `can_hold_element` actually fully covers what I needed here. 

It only doesn't do `extract_array` compared to `Block._can_hold_element`, so kept that here for now. But probably that could be moved to `can_hold_element` as well (will check later).